components/builder-depot-client/src/lib.rs

```diff
@@ -779,35 +783,43 @@ impl Client {
         &self,
         pa: &mut PackageArchive,
         token: &str,
+        force_upload: bool,
         progress: Option<D>,
     ) -> Result<()>
     where
         D: DisplayProgress + Sized,
     {
         let checksum = pa.checksum()?;
         let ident = pa.ident()?;
-        let mut file =
-            File::open(&pa.path).map_err(|e| Error::PackageReadError(pa.path.clone(), e))?;
-        let file_size = file.metadata()
+        let file = File::open(&pa.path).map_err(|e| Error::PackageReadError(pa.path.clone(), e))?;
+        let file_size = file
+            .metadata()
             .map_err(|e| Error::PackageReadError(pa.path.clone(), e))?
             .len();
+
         let path = package_path(&ident);
+
+        let forced = if force_upload { "true" } else { "false" };
```

Comment:
You don't need to do this... just do a .to_string() when passing it to append_pair

Reply:
Oh yeah I didn't even think about the fact that I'm passing this in as a string 100% of the time now. Good catch.